wgpu-native/src/device.rs

```diff
@@ -310,30 +320,33 @@ impl DestroyedResources<back::Backend> {
 
 fn map_buffer(
     raw: &<back::Backend as hal::Backend>::Device,
+    limits: &hal::Limits,
     buffer: &mut resource::Buffer<back::Backend>,
-    range: std::ops::Range<u64>,
-    read: bool,
-    write: bool,
+    mut range: std::ops::Range<u64>,
+    kind: HostMap,
 ) -> Result<*mut u8, hal::mapping::Error> {
+    let is_coherent = buffer.memory_properties.contains(hal::memory::Properties::COHERENT);
+    let atom_mask = if is_coherent { 0 } else { limits.non_coherent_atom_size as u64 - 1 };
```

Comment:
minor: non_coherent_atom_size might be 0? Is it guaranteed to be pow of 2?

Reply:
Since stuff needs to be a multiple of it according to the spec, I don't expect it to be 0.
Also - https://vulkan.gpuinfo.org/displaydevicelimit.php?name=nonCoherentAtomSize